client/my-sites/comments/controller.js

```diff
@@ -15,67 +15,100 @@ import route, { addQueryArgs } from 'lib/route';
 import { getNotices } from 'state/notices/selectors';
 
-const VALID_STATUSES = [ 'all', 'pending', 'approved', 'spam', 'trash' ];
+const mapPendingStatusToUnapproved = status => ( 'pending' === status ? 'unapproved' : status );
 
-export const isValidStatus = status => includes( VALID_STATUSES, status );
-
-export const getRedirectUrl = ( status, siteFragment ) => {
-	const statusValidity = isValidStatus( status );
-	if ( status === siteFragment ) {
-		return `/comments/all/${ siteFragment }`;
-	}
-	if ( ! statusValidity && ! siteFragment ) {
-		return '/comments/all';
-	}
-	if ( ! statusValidity && siteFragment ) {
-		return `/comments/all/${ siteFragment }`;
-	}
-	if ( statusValidity && ! siteFragment ) {
-		return `/comments/${ status }`;
+const changePage = path => pageNumber => {
+	if ( window ) {
+		window.scrollTo( 0, 0 );
 	}
-	return null;
+	return page( addQueryArgs( { page: pageNumber }, path ) );
 };
 
-export const redirect = function( context, next ) {
-	const { status, site } = context.params;
-	const siteFragment = route.getSiteFragment( context.path );
-	const redirectUrl = getRedirectUrl( status, siteFragment );
-	if ( redirectUrl && ( site || '/comments/' + status !== redirectUrl ) ) {
-		return page.redirect( redirectUrl );
+export const siteComments = ( { params, path, query, store } ) => {
+	const siteFragment = route.getSiteFragment( path );
+
+	if ( ! siteFragment ) {
+		return page.redirect( '/comments/all' );
 	}
-	next();
-};
 
-const changePage = ( status, siteFragment ) => pageNumber => {
-	if ( window ) {
-		window.scrollTo( 0, 0 );
+	const status = mapPendingStatusToUnapproved( params.status );
+
+	const pageNumber = parseInt( query.page, 10 );
+	if ( isNaN( pageNumber ) || pageNumber < 1 ) {
+		return changePage( path )( 1 );
 	}
 
-	return page( addQueryArgs( { page: pageNumber }, `/comments/${ status }/${ siteFragment }` ) );
+	renderWithReduxStore(
+		<CommentsManagement
+			basePath={ path }
+			changePage={ changePage( path ) }
+			page={ pageNumber }
+			siteFragment={ siteFragment }
+			status={ status }
+		/>,
+		'primary',
+		store
+	);
 };
 
-export const comments = function( context ) {
-	const status = 'pending' === context.params.status ? 'unapproved' : context.params.status;
-	const siteFragment = route.getSiteFragment( context.path );
+export const postComments = ( { params, path, query, store } ) => {
+	const siteFragment = route.getSiteFragment( path );
+
+	if ( ! siteFragment ) {
+		return page.redirect( '/comments/all' );
+	}
+
+	const status = mapPendingStatusToUnapproved( params.status );
+	const postId = parseInt( params.post, 10 );
 
-	const pageNumber = parseInt( context.query.page, 10 );
-	if ( isNaN( pageNumber ) || pageNumber === 0 ) {
-		return changePage( context.params.status, siteFragment )( 1 );
+	if ( ! isFinite( postId ) ) {
+		return page.redirect( `/comments/${ params.status }/${ siteFragment }` );
+	}
+
+	const pageNumber = parseInt( query.page, 10 );
```

Comment:
We could refactor these page number validation steps into helper function to avoid repeating https://github.com/Automattic/wp-calypso/pull/19007/files#diff-5ab0c8264a5b0a1fda0f7eb5c9246212R36.